### PR TITLE
Add image pull secret

### PR DESCRIPTION
**Pull request by junaid18183**

### CODE REVIEW

File: helm/kiam/README.md [hunk 1]
```diff
@@ -168,6 +168,7 @@ Parameter | Description | Default
 `server.useHostNetwork` | If true, use hostNetwork on server to bypass agent iptable rules | `false`
 `rbac.create` | If `true`, create & use RBAC resources | `true`
 `psp.create` | If `true`, create Pod Security Policies for the agent and server when enabled | `false`
+`imagePullSecret` | The name of the secret to use if pulling from a private registry | `nil`
```
[the remark this answers is not in the document]
Comment (rhysemmas) [reply helm/kiam/README.md]: Also update `imagePullSecret` to `imagePullSecrets` (see `values.yaml` comments)

**Comment by rhysemmas** [comment]
Hi @junaid18183,

Many thanks for your contribution - please let me know your thoughts on my comments.

File: helm/kiam/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: kiam
-version: 3.0.1
+version: 3.0.2
```
[the remark this answers is not in the document]
Comment (junaid18183) [reply helm/kiam/Chart.yaml]: Agree, Updating the minor version.

File: helm/kiam/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (junaid18183) [reply helm/kiam/README.md]: Fixed

File: helm/kiam/templates/agent-daemonset.yaml [hunk 3]
```diff
@@ -134,6 +134,10 @@ spec:
           resources:
 {{ toYaml .Values.agent.resources | indent 12 }}
         {{- end }}
+        {{- if .Values.imagePullSecret }}
+      imagePullSecrets:
+        - name: {{ .Values.imagePullSecret }}
```
[the remark this answers is not in the document]
Comment (junaid18183) [reply helm/kiam/templates/agent-daemonset.yaml]: Fixed

File: helm/kiam/templates/server-daemonset.yaml [hunk 4]
```diff
@@ -140,6 +140,10 @@ spec:
           resources:
 {{ toYaml .Values.server.resources | indent 12 }}
         {{- end }}
+        {{- if .Values.imagePullSecret }}
+      imagePullSecrets:
+        - name: {{ .Values.imagePullSecret }}
```
[the remark this answers is not in the document]
Comment (junaid18183) [reply helm/kiam/templates/server-daemonset.yaml]: Fixed

File: helm/kiam/values.yaml [hunk 5]
```diff
@@ -1,5 +1,8 @@
 extraArgs: {}
 
+## The name of the secret to use if pulling images from a private registry.
+# imagePullSecret: test
```
[the remark this answers is not in the document]
Comment (junaid18183) [reply helm/kiam/values.yaml]: Agree, fixed

**Comment by junaid18183** [comment]
Please have a look

**Comment by rhysemmas** [comment]
LGTM - thanks for the changes, please resolve the conflicts and then we can merge

**Comment by junaid18183** [comment]
Done, please check now

**Pull request merged**